### ISSUE TITLE
Calculate average and low_price_cutoff from all available prices

### ISSUE THREAD

**Issue by bjornbak** [custom-components/nordpool#99]
**Is your feature request related to a problem? Please describe.**
I got a Sonnen solar PV setup with a 15 kWh battery.
I see the prices in DK2 varying wildly with sometimes a 300+% difference between off-peak hours and max hours.
I am using low_price_cutoff to decide whether to charge the battery in off-peak hours to use in the the high priced hours.
The lowest priced off-peak hours are typically from 01:00 to 06:00 but sometimes after 21:00 are below the threshold too.

But then to the problem:
- the prices are also varying from day to day so off-peak after 21:00 today might be above the low_price_cutoff if I include tomorrow's average and then I want to wait with charging.
- Or the prices tomorrow might be so high that even if after 21:00 isn't below low_price_cutoff today it is still cheaper than charging after 00:00...

**Describe the solution you'd like**
I would like it to use all available prices to calculate the average for low_price_cutoff instead of the per day average.

**Describe alternatives you've considered**
I could contribute with a "rolling_low_price_cutoff" PR if you find a good idea or I can implement it locally in my own HA.

**Additional context**

**Comment by Hellowlol** [comment]
Hi Christian! You can make a PR for it 👍

**Comment by ThoStaKa** [comment]
@bjornbak This comment is out of this integration's scope, but you could check out Node Red Power Saver. This tool takes "all" future prices into account for consumption planning. It uses input from Nordpool (and other sources if you like)
[https://flows3.nodered.org/node/node-red-contrib-power-saver](url)

**Comment by bjornbak** [comment]
@ThoStaKa not at all out of scope :) Why spend time implementing stuff that already exists!?
Luckily I haven't had time to look at it yet so I'll take a look at Node Red Power Saver before starting..

**Comment by foolip** [comment]
I'm also interested in consumption planning. From a quick read of https://github.com/ottopaulsen/node-red-contrib-power-saver/blob/main/src/strategy-best-save-functions.js it looks like that code does implement the kind of logic I've been thinking of, to delay heating by some amount.

However, I'm not thrilled about lots of dependencies. Does anyone on this thread know of any home assistant integrations or custom components that create an on/off schedule like this?

**Comment by bjornbak** [comment]
@foolip @ThoStaKa I finally got around to look at node-red-contrib-power-saver today but it doesn't solve my need.
My need is to look at the average today and tomorrow (if available) and find hours where the price is low to allow my solar system to charge with grid power to have some in spare when the price are high. I could get node-red-contrib-power-saver as it still insisted on charging today as well even when the prices is 6x tomorrow's.

I see here in area DK2 wildly varying prices so the off-peak2 prices might be higher than the following off-peak1 or vice-versa even if they are connected at midnight. And tomorrow the prices are even more insane with off-peak prices 6x the peak prices... Go figure.. 
![image](https://user-images.githubusercontent.com/79479/149637243-1813d164-a9b3-4ef5-bb92-90e7aa04c393.png)

The nordpool component's average doesn't help because it simply transfer the per-day average from the nordpool feed.
Inspired by node-red-contrib-power-saver I ended up custom implementing it in node red (using this component to read nordpool) 
But I am sure it can be implemented in automations or the nordpool component...
